### PR TITLE
Add dishes to cart

### PR DESCRIPTION
**Pull request by ceca2017**
### Add dishes to cart
https://www.pivotaltracker.com/story/show/145100535

* Use cookies to store the dishes (using session feature in code) in a cart

### CODE REVIEW

File: features/dishes_to_cart.feature [hunk 1]
```diff
@@ -0,0 +1,31 @@
+Feature: Add dishes to menu
+  As a visitor
+  In order to select a dish that I´d like to order
+  I would like to place it in the cart
+
+  Background:
+  Given "benji@teabags.com" exists and has 2 restaurants belonging to 2 restaurant categories
```
Comment (diraulo) [inline features/dishes_to_cart.feature]: indentation on this line

File: app/controllers/carts_controller.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+class CartsController < ApplicationController
+  def index
```
Comment (diraulo) [inline app/controllers/carts_controller.rb]: `Index` doesn't feel like the right controller action for this. I'd expect it to be in an action called `add_to_cart` maybe ~~or `create`?~~ 🤔

File: app/controllers/carts_controller.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+    session[:cart] ||= []
+    dish = Dish.find(params[:dish_id])
+    @restaurant_id = params[:restaurant_id]
+    if session[:cart] << dish
+      redirect_to restaurant_menus_path(@restaurant_id)
+      flash[:notice] = "#{dish.name} added to cart"
+    else
+      flash[:notice] = "Error: #{dish.name} was NOT added to cart"
```
Comment (diraulo) [inline app/controllers/carts_controller.rb]: Why are we implementing the sad path already when we have no tests requiring this? Stick to your tests and only implement what's required to get them to pass.

File: app/views/menus/index.html.haml [hunk 4]
```diff
@@ -6,3 +6,5 @@
     %li= dish.name
     %li= dish.description
     %li= dish.price
+    %li= link_to "Add #{dish.name} to cart", carts_path(:dish_id => dish, :restaurant_id => @restaurant.id)
```
Comment (diraulo) [inline app/views/menus/index.html.haml]: don't use hash-rockets for hashes 🙄

**Review by diraulo — DISMISSED** [review]

**Comment by andycarnahan** [comment]
It looks like you've addressed everything. Update the branch, fix any conflicts, and I'll merge it.

**Review by andycarnahan — APPROVED** [review]
All the comments brought up by Raoul have been addressed, so I'm approving this.

**Pull request merged**